src/Common/Config/AbstractConfigurationComparison.h

```diff
@@ -33,6 +34,11 @@ namespace DB
     bool isSameConfiguration(const Poco::Util::AbstractConfiguration & left, const String & left_key,
                              const Poco::Util::AbstractConfiguration & right, const String & right_key);
 
+    /// Returns true if specified subviews of the two configurations contains the same keys and values, but without checking specified keys.
+    bool isSameConfigurationIgnoringKeys(const Poco::Util::AbstractConfiguration & left, const String & left_key,
```

Comment:
isSameConfiguration & isSameConfigurationIgnoringKeys can be combined into one function making the last parameter default to empty set. what do you think?

Reply:
I think it's probably better to keep two functions for now, because in most cases we don't need to ignore keys.

Reply:
done